### DISCUSSION TITLE
Embed SumatraPDF into a medical device

### DISCUSSION THREAD

**Discussion by sysopmd** [sumatrapdfreader/sumatrapdf#3977] [Q&A]
Hi all,
I tried researching the topic a litte, but can't come up with a comprehensive understanding, so I thought I'd ask here.

I recently discovered SumatraPDF and I appreciate its high flexibility and customizability, which would make it ideal for integration into our medical device as a way to display the device's user manual.

I understand that the SW is GPLv3-licensed. We are not planning to perform any modifications to its sources, so my understanding is that we can include it in our device, provided that:
- we inform our user that we are using SumatraPDF
- provide a link to the website where they could download the source code

However, I also found out that GPLv3 has a requirement which may seem reasonable for some consumer product but I find very difficult to comply with for medical devices: the so-called [User Product Clause](https://fossa.com/blog/complying-gpl-v3s-user-product-clause/).

As manufacturers, we need to ensure that we tested our product with an exact version of every SW component we include, and we need to certify and/or notify the certification body for any changes or SW upgrades. There is no way we can let our users upgrade a piece of the SW autonomously and still guarantee that the device is safe and effective for our end users.

On the other hand, a product such as SumatraPDF seems to be designed exactly for cases where it can be bundled to some other application, so I am sure we are not the first ones facing this issue.

Is there an obvious solution that I am missing here?
Is there a way to license the product with a different licensing maybe? or are there exception to this clause?
What about the used libraries?

Thanks in advance for any insight or suggestion

**Comment by GitHubRulesOK** [comment]
I am not the developer so these are personal opinions (IMHO)
To move away from default behaviour (constantly updating)

 i..e. standalone .exe is used without any guarantee as to purpose and is thus NOT designed for life critical applications
>This program is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE. See the GNU General Public License for more details.

You would need to consider Artifex Commercial License for MuPDF core application as an alternative, however their product is likely to carry similar avoidance of "fit for intended use" as the open License is standard AGPL

**Reply by sysopmd** [reply]
Thanks for the reply.
It is of course the manufacturer's responsibility to check that the product is "fit for intended use".
Anyways, our medical device is not "life threatening", but for skin aesthetic treatments. Nonetheless, it should comply to medical device regulations.

As for the Artifex Commercial License: thanks for the suggestion. We can of course consider to license that, but this will cover only MuPDF library, right? do you know if SumatraPDF also have such options?

**Reply by GitHubRulesOK** [reply]
For a definitive answer you could email the developer (@kjk) , but the overarching principle is SumatraPDF is composed primarily of MuPDF viewing code (based on its licenced sub component permits) with additional permitted User Interface embellishments.

**Reply by sysopmd** [reply]
Thanks, I'll write to him.

**Comment by kjk** [comment]
I'm not a lawyer but as the creator of SumatraPDF: don't worry about user product clause (in the context of SumatraPDF).

Feel free to use SumatraPDF in your product without giving users ways to update.
